src/test/java/net/kaczmarzyk/spring/data/jpa/web/SimpleSpecificationResolverTest.java

```diff
@@ -273,6 +273,50 @@ public void buildsTheSpecUsingMultiValueWebParameterAndParamSeparator() throws E
         assertThat(resolved).isEqualTo(new In<>(queryCtx, "thePath", new String[] { "val1", "val2", "val3", "val4", "val5", "val6", "val7" }, converter));
     }
 
+    @Test(expected = IllegalStateException.class)
+    public void throwsIllegalStateExceptionWhenMethodWasNotFound() {
+        MethodParameter param = MethodParameter.forExecutable(testMethod("testMethod9"), 0);
+        NativeWebRequest req = mock(NativeWebRequest.class);
+
+        when(req.getParameterValues("theParameter")).thenReturn(new String[] {"val1", "val2,val3,val4", "val5,val6", "val7"});
+
+        WebRequestProcessingContext ctx = new WebRequestProcessingContext(param, req);
+
+        resolver.buildSpecification(ctx, param.getParameterAnnotation(Spec.class));
+    }
+
+    @Test
+    public void shouldReturnTrueIfAnnotationIsSupportedByResolver() {
+        MethodParameter param = MethodParameter.forExecutable(testMethod("testMethod10"), 0);
+
+        assertThat(resolver.supports(param.getParameterAnnotations()[0])).isTrue();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void shouldThrowIllegalStateExceptionWhenDefConfigLengthIsMoreThanOne(){
+        MethodParameter param = MethodParameter.forExecutable(testMethod("testMethod11"), 0);
+        NativeWebRequest req = mock(NativeWebRequest.class);
+
+        when(req.getParameterValues("theParameter")).thenReturn(new String[] {"val1", "val2,val3,val4", "val5,val6", "val7"});
+
+        WebRequestProcessingContext ctx = new WebRequestProcessingContext(param, req);
+
+        resolver.buildSpecification(ctx, param.getParameterAnnotation(Spec.class));
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void shouldThrowIllegalStateExceptionWhenExceptionOccurs(){
+        MethodParameter param = MethodParameter.forExecutable(testMethod("testMethod11"), 0);
+        NativeWebRequest req = mock(NativeWebRequest.class);
+
+        when(req.getParameterValues("theParameter")).thenReturn(new String[] {"val1", "val2,val3,val4", "val5,val6", "val7"});
```

Comment:
why exception is expected in this test? refactor code or at least add an explanation in a comment

Reply:
deleted this test method, i accidently abandoned it without refactoring. the goal of this method was to cover the last catch of this try-catch clause:
```
} catch (NoSuchMethodException e) {
			throw new IllegalStateException("Does the specification class expose at least one of the supported constructors?\n"
					+ "It can be either:\n"
					+ "  3-arg (QueryContext queryCtx, String path, String[] args)\n"
					+ "  4-arg (QueryContext queryCtx, String path, String[] args, Converter converter)\n"
					+ "  5-arg (QueryContext queryCtx, String path, String[] args, Converter converter, String[] config)", e);
		} catch (RuntimeException e) {
			throw e;
		} catch (Exception e) {
			throw new IllegalStateException(e);
		}
```
imo this edge case should not happen and can be invoked only by throwing exception by mock. i think that it doesn't make sense to test it.